test/parallel/test-module-create-require-from-directory.js

```diff
@@ -0,0 +1,18 @@
+'use strict';
+
+require('../common');
+const assert = require('assert');
+const path = require('path');
+
+const { createRequireFromPath } = require('module');
+
+const fixPath = path.resolve(__dirname, '..', 'fixtures');
+const p = path.join(fixPath, path.sep);
+
+const req = createRequireFromPath(p);
+const reqFromNotDir = createRequireFromPath(fixPath);
+
+assert.strictEqual(req('./baz'), 'perhaps I work');
+assert.throws(() => {
+  reqFromNotDir('./baz');
+}, { code: 'MODULE_NOT_FOUND' });
```

Comment:
I'd expect both of this to pass. `fixtures`  is a directory, so they should both work IMHO. This probably requires adding a round-trip to OS, but I don't think it could be avoided.

Reply:
That would be a very different approach to this function, but it could work. //cc @devsnek

Reply:
I'm happy to change the implementation to call `stat.isDirectory()` if that's the way we want to move this forward :)

Reply:
If we want to get this merged, I'd suggest we stick with the current behaviour which matches the behaviour of path resolution.

Reply:
+1 to sticking to behavior that matches path... we should change this behavior in the path module if we want to fix it imho

Reply:
I personally think calling `stat` is a good idea in this case. `createRequireFromPath` is IMO special enough to work differently from the rest of `Module`.

Reply:
I would like to land this to improve the behavior and introduce the stat separately in a follow up. Is that reasonable for you @BridgeAR ?

Reply:
This has three different open comments at the moment (actually four but two are about using `stat`). I personally would not land this but I won't block it either.

Reply:
@BridgeAR afaict all open comments are about stat, which I think should be an incremental improvement discussed separately.

Did I miss any other open comments?